qiskit_experiments/framework/composite/composite_analysis.py

```diff
@@ -44,31 +45,69 @@ class CompositeAnalysis(BaseAnalysis):
         reconstructed from the parent composite experiment data.
     """
 
+    def __init__(self, analyses: List[BaseAnalysis]):
+        """Initialize a composite analysis class.
+
+        Args:
+            analyses: a list of component experiment analysis objects.
+        """
+        super().__init__()
+        self._analyses = analyses
+
+    def component_analysis(self, index=None) -> Union[BaseAnalysis, List[BaseAnalysis]]:
+        """Return the component experiment Analysis object"""
+        if index is None:
+            return self._analyses
+        return self._analyses[index]
+
     def _run_analysis(self, experiment_data: ExperimentData):
+        # Return list of experiment data containers for each component experiment
+        # containing the marginalied data from the composite experiment
+        component_exp_data = self._component_experiment_data(experiment_data)
+
+        # Run the component analysis on each component data
+        for sub_exp_data, sub_analysis in zip(component_exp_data, self._analyses):
+            # Since copy for replace result is handled at the parent level
+            # we always run with replace result on component analysis
+            sub_analysis.run(sub_exp_data, replace_results=True)
```

Comment:
There are already a lot of composite experiment tests but i think they all use the same component experiment, it would be good to add some tests that mix different experiments like you suggest